internal/provider/resources/workspace_role.go

```diff
@@ -0,0 +1,375 @@
+		return
+	}
+
+	client, ok := req.ProviderData.(api.PrefectClient)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected api.PrefectClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *WorkspaceRoleResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Resource representing a Prefect Workspace Role",
+		Version:     0,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "Workspace Role UUID",
+				// attributes which are not configurable + should not show updates from the existing state value
+				// should implement `UseStateForUnknown()`
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"created": schema.StringAttribute{
+				Computed:    true,
+				CustomType:  customtypes.TimestampType{},
+				Description: "Date and time of the Workspace Role creation in RFC 3339 format",
+			},
+			"updated": schema.StringAttribute{
+				Computed:    true,
+				CustomType:  customtypes.TimestampType{},
+				Description: "Date and time that the Workspace Role was last updated in RFC 3339 format",
+			},
+			"name": schema.StringAttribute{
+				Required:    true,
+				Description: "Name of the Workspace Role",
+			},
+			"description": schema.StringAttribute{
+				Optional:    true,
+				Description: "Description of the Workspace Role",
+			},
+			"scopes": schema.ListAttribute{
+				Description: "List of scopes linked to the Workspace Role",
+				ElementType: types.StringType,
+				Optional:    true,
+			},
+			"account_id": schema.StringAttribute{
+				CustomType:  customtypes.UUIDType{},
+				Description: "Account UUID, defaults to the account set in the provider",
+				Computed:    true,
+			},
+			"inherited_role_id": schema.StringAttribute{
+				CustomType:  customtypes.UUIDType{},
+				Description: "Workspace Role UUID, whose permissions are inherited by this Workspace Role",
+				Optional:    true,
+			},
+		},
+	}
+}
+
+// copyWorkspaceRoleToModel copies an api.WorkspaceRole to a WorkspaceRoleDataSourceModel.
+func copyWorkspaceRoleToModel(ctx context.Context, role *api.WorkspaceRole, model *WorkspaceRoleResourceModel) diag.Diagnostics {
+	model.ID = types.StringValue(role.ID.String())
+	model.Created = customtypes.NewTimestampPointerValue(role.Created)
+	model.Updated = customtypes.NewTimestampPointerValue(role.Updated)
+
+	model.Name = types.StringValue(role.Name)
+	model.Description = types.StringPointerValue(role.Description)
+	model.AccountID = customtypes.NewUUIDPointerValue(role.AccountID)
+	model.InheritedRoleID = customtypes.NewUUIDPointerValue(role.InheritedRoleID)
+
+	// NOTE: here, we'll omit updating the TF state with the scopes returned from the API
```

Comment:
let's say a practitioner configures a `manage_work_pools` scope in terraform.  Prefect Cloud will create the role + automatically add subordinate scopes, like `see_work_pools` and `write_work_pools`.  This will always cause a conflict on apply, as the practitioner will only have defined `manage_work_pools`

in this case, it's ok to hold onto the user-defined parameters, as they'll more like request inputs vs. a full declarative reflection of the scopes assigned to this role